test/forms/mappings/MappingsSpec.scala

```diff
@@ -38,9 +38,98 @@ object MappingsSpec {
 }
 
 class MappingsSpec extends AnyFreeSpec with Matchers with OptionValues with Mappings {
-
   import MappingsSpec._
 
+  "postcode" - {
+
+    val testForm: Form[String] =
+      Form(
+        "value" -> postcode)
+
+    "must bind a valid string that matches postcode regex" - {
+      "with format AA1 1AA - spaces" in {
+        val result = testForm.bind(Map("value" -> "AA1 1AA"))
+        result.get mustEqual "AA1 1AA"
+      }
+
+      "with format AA11 1AA - spaces" in {
+        val result = testForm.bind(Map("value" -> "AA11 1AA"))
+        result.get mustEqual "AA11 1AA"
+      }
+
+      "with format A1 1AA - spaces" in {
+        val result = testForm.bind(Map("value" -> "A1 1AA"))
+        result.get mustEqual "A1 1AA"
+      }
+
+      "with format A11 1AA - spaces" in {
+        val result = testForm.bind(Map("value" -> "A11 1AA"))
+        result.get mustEqual "A11 1AA"
+      }
+
+      "with format AA1 1AA - no spaces" in {
+        val result = testForm.bind(Map("value" -> "AA11AA"))
+        result.get mustEqual "AA1 1AA"
+      }
+
+      "with format AA11 1AA - no spaces" in {
+        val result = testForm.bind(Map("value" -> "AA111AA"))
+        result.get mustEqual "AA11 1AA"
+      }
+
+      "with format A1 1AA - no spaces" in {
+        val result = testForm.bind(Map("value" -> "A11AA"))
+        result.get mustEqual "A1 1AA"
+      }
+
+      "with format A11 1AA - no spaces" in {
+        val result = testForm.bind(Map("value" -> "A111AA"))
+        result.get mustEqual "A11 1AA"
+      }
+
+      "with format BFPO 11111 - spaces" in {
+        val result = testForm.bind(Map("value" -> "BFPO 11111"))
+        result.get mustEqual "BFPO11111"
+      }
+
+      "with format BFPO 11111 - no spaces" in {
+        val result = testForm.bind(Map("value" -> "BFPO11111"))
+        result.get mustEqual "BFPO11111"
+      }
+    }
+
+    "must not bind a string that does not match postcode regex" in {
```

Comment:
For all errors on this page, my suggestion would be to type out the expected error.  If somewhere were to inadvertently delete or change something in messages, this would rightly cause failing tests to alert us that something has changed.